### ISSUE TITLE
AsyncAPI specification

### ISSUE THREAD

**Issue by fmvilas** [zalando/restful-api-guidelines#430]
Hi! I just came across your API guidelines and found it awesome. And what I liked the most is that you also document your events using OpenAPI schemas. I just wanted to know if you're aware of the [AsyncAPI specification](https://github.com/asyncapi/asyncapi). It seems like a good fit for your use case and its schemas are exactly the same as OpenAPI schemas.

**Comment by tkrop** [comment]
@fmvilas thanks for sharing this information with us. We will have a look to it.

**Comment by fmvilas** [comment]
Cool! Make sure you try editor.asyncapi.org. It will be easier.

**Comment by vadeg** [comment]
Hi @fmvilas!

Thank you for sharing. We have discussed it and decided to try to apply this specification to some of our solutions. Will see how it fits.

**Comment by fmvilas** [comment]
That's great news! Let me know if you need help. We're currently setting up a core team for the AsyncAPI specification. Whatever questions you have, let me know here or [on the Slack channel](https://async-apis-slack.herokuapp.com). We'll be glad to help!

**Comment by fmvilas** [comment]
If interested in generating Markdown docs, check out this package: https://github.com/asyncapi/generator/blob/master/README.md

**Comment by vadeg** [comment]
Research showed that AsyncAPI schema can be used with our services. 

I have tested it with Nakadi event types. Event types schema almost matches to AsyncAPI with some exception. Nakadi schema support more data types rather than AsyncAPI schema. As far as I understand this can be extended.

There is no `http/https` scheme support. This is not a blocked and [will be fixed soon](https://github.com/asyncapi/asyncapi/issues/81).

There are several with parser which does not allow to build schemes using `streams`. Only schema with `topic` works fine. There is also some problems with error notifications in online editor (the problem also comes from parser).

In general I think AsyncAPI schema can be applied.

P.S. @tkrop please update this ticket if I missed some points.

**Comment by vadeg** [comment]
Version 1 of Async API has been investigated and tested. There were some questions regarding fields semantics and limitations in Async API v1 such as:
1. Rename `scheme` field in [Server Object](https://github.com/asyncapi/asyncapi/blob/master/versions/1.2.0/asyncapi.md#server-object) to `protocol` because `protocol` describes better what is this field actually about.
2. Allow to specify custom protocols in `scheme` field. Async API v1 allows to specify protocols only from predefined list.

A [feature request](https://github.com/asyncapi/asyncapi/issues/214) was created do discuss questions mentioned above. @fmvilas pointed that there will be version 2 released soon which solves second issue. First issue have been solved in v2 also.

In summary, I would recommend to wait until Async API v2 will be released, refine our internal document and discuss it again.

**Comment by tfrauenstein** [comment]
here, the summary document of our discussion based on AsyncAPI V1
https://docs.google.com/document/d/1MJRFZwMTwjWE-cz5AFYOTqaN96fqf8OLNAunp2IhKVg/edit#heading=h.l9hrm1wom4lh

**Comment by ePaul** [comment]
**Guild Meeting 2019-09-10**

* we looked into the changes for AsyncAPI 2.0 – mostly changing topic→channel
* it could possibly be used for Nakadi event types by adding another binding

Possible approach:
* define event types in AsyncAPI
* generate Nakadi event type definitions from that

Alternative:
* convert Nakadi event types to AsyncAPI
* integrate those into the API portal

But:
* more tooling needed
* we don't see yet any major industry player adopting this (except Slack?)
* It currently doesn't really offer a big benefit – a generic client (or client generator) across multiple protocols (e.g. Nakadi, SQS, ...) might be helpful here

We'll discuss with the Nakadi team how to proceed here.

**Comment by ePaul** [comment]
Waiting until after Black Friday.

**Comment by zeitlinger** [comment]
> Waiting until after Black Friday.

the answer to life the universe and everything

**Comment by ePaul** [comment]
**Meeting at 2020-02-11**

We currently can't give any priorities to this topic (e.g. Nakadi integration), so we are closing it for now.
